### DISCUSSION TITLE
Add Protection to Naming Villagers

### DISCUSSION THREAD

**Discussion by Tekniqality** [GriefPrevention/GriefPrevention#1333] [Ideas]
Currently Villagers who are inside claims, can be renamed by anyone (whether they have trust or not). Would be nice to have an option to disable this (even if its just a config setting)

**Comment by RoboMWM** [comment]
Does this apply to other mobs too?

**Reply by Tekniqality** [reply]
As far as I’m aware all passive mobs are protected from being renamed/harmed while in claims except for villagers. I’m also not completely sure about hostile mobs (zombies in natural IG farms for example)

**Reply by RoboMWM** [reply]
That's strange. Not sure why villagers would be missing from that protection...

**Reply by Jikoo** [reply]
I assume it's because GP doesn't specifically check for renaming at any point, only interaction. If trading is allowed, renaming probably is also allowed.

**Reply by Tekniqality** [reply]
Ahh that’s a possibility so then if trading was disabled would that stop others from renaming?

**Reply by Jikoo** [reply]
It should, yeah.

**Reply by Tekniqality** [reply]
> That's strange. Not sure why villagers would be missing from that protection...

Assuming that what was mentioned is the case, Any chance of getting it as a seperate config setting? Just so it’s possible to have public trading halls without the worry of people griefing the names

**Reply by Jikoo** [reply]
Seems like it shouldn't need a config, should probably be tied to the ability to kill. Kind of a mess to work into GP's existing logic though.
